src/TIN.php

```diff
@@ -120,6 +120,17 @@ public function isValid(bool $strict = false): bool
         return true;
     }
 
+    public function hasAlgorithm(): bool
```

Comment:
How about keeping the "algorithm" side of things internally and expose a simpler interface?

In this case, I would just name this method `isSupported()` ? WDYT ?

Reply:
Yes, this is better, I agree you, better isSupported()